code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -273,14 +273,51 @@
 		ITEM_SLOT_LPOCKET, ITEM_SLOT_RPOCKET,\
 		ITEM_SLOT_DEX_STORAGE
 	)
+	var/kicked_over = FALSE
 
-/obj/item/reagent_containers/glass/bucket/wooden
-	name = "wooden bucket"
-	icon_state = "woodbucket"
-	inhand_icon_state = "woodbucket"
-	custom_materials = list(/datum/material/wood = MINERAL_MATERIAL_AMOUNT * 2)
-	armor = list(MELEE = 10, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, FIRE = 0, ACID = 50)
-	resistance_flags = FLAMMABLE
+/obj/item/reagent_containers/glass/bucket/Initialize(mapload)
+	. = ..()
+	update_appearance()
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = .proc/kicking_the_bucket,
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/item/reagent_containers/glass/bucket/proc/bucket_got_kicked()
```

Comment:
doc this proc!

Reply:
Done. I documented it as briefly as possible but I can give more details if that would be better.